examples/language-modeling/main.py

```diff
@@ -162,7 +166,51 @@
     torch_device = torch.device(device_str)
 
     is_glm = bool(re.search("chatglm", model_name.lower()))
-    if is_glm:
+    if args.layer_wise:
+        from auto_round.layer_wise.utils import load_model_with_hooks
+        if is_glm:
+            model = load_model_with_hooks(
+                model_name,
+                AutoModel,
+                device=device_str,
+                clean_weight=True,
+                saved_path="./layer_wise_tmp",
+                torch_dtype=torch_dtype,
+                trust_remote_code=not args.disable_trust_remote_code
+                )
+        else:
+            model = load_model_with_hooks(
+                model_name,
+                AutoModelForCausalLM,
+                device=device_str,
+                clean_weight=True,
+                saved_path="./layer_wise_tmp",
```

Comment:
save_path should be in the output_dir, if not set, then using working directory. Remeber to delete this tmp file